zarr/storage.py

```diff
@@ -946,6 +947,135 @@ def atexit_rmglob(path,
             rmtree(p)
 
 
+class FSStore(MutableMapping):
+
+    def __init__(self, url, normalize_keys=True, key_separator='.',
+                 mode='w', consolidated=False, metadata_key='.zmetadata',
+                 exceptions=(KeyError, PermissionError, IOError),
+                 **storage_options):
+        import fsspec
+        self.path = url
+        self.normalize_keys = normalize_keys
+        self.key_separator = key_separator
+        self.map = fsspec.get_mapper(url, **storage_options)
+        self.fs = self.map.fs  # for direct operations
+        self.mode = mode
+        self.exceptions = exceptions
+        # TODO: should warn if consolidated and write mode?
+        if self.fs.exists(url) and not self.fs.isdir(url):
+            err_fspath_exists_notdir(url)
+        self.consolidated = consolidated
+        self.metadata_key = metadata_key
+        if consolidated:
+            self.meta = json.loads(self.map.get(metadata_key, b"{}").decode())
```

Comment:
There is a `json_loads` utils function used in other place of this file:

```
def json_loads(s):
    """Read JSON in a consistent way."""
    return json.loads(ensure_text(s, 'ascii'))
```

Do you want to use it ?